### ISSUE TITLE
Support sharded multi-tenant database model

### ISSUE THREAD

**Issue by fabiofdsantos** [Errorname/prisma-multi-tenant#31]
First of all, thanks for this amazing work!

Since this project is using "database per tenant" as tenancy model, are you considering to support the shared multi-tenant model in the future?

More details:
https://github.com/microsoft/WingtipTicketsSaaS-MultiTenantDB#sharded-multi-tenant-saas-pattern

**Comment by Errorname** [comment]
Hi, thank you for the compliment 🙂 

The initial purpose of this project is to be a wrapper of Prisma to handle the multi-tenant design pattern. I like the fact that `prisma-multi-tenant` does not touch your schema, nor your databases, but only act as a "router" to find your data based on the tenant. You could even "eject" of `prisma-multi-tenant` simply by uninstalling the package.

For what I understand, supporting the sharded multi-tenant pattern would require to have complete control over the schema of the project using `prisma-multi-tenant`, and making important changes in the schema. This would greatly increase the complexity of the package.

I'm not against such evolution, however, I'll probably not do it in the near future, as that would require a lot of work.

However, if you have an idea on how I could implement this pattern in a way that would:
- not increase the overall complexity too much, and
- not be invasive to the projects using `prisma-multi-tenant`,
I'd love to hear about it! 🙂

**Comment by fabiofdsantos** [comment]
I agree with you that It will lead to more complexity and currently I'm not looking for that solution but maybe a "nice to have" in the future since the pattern has some advantages.

However database-per-tenant model will be the best approach for most applications. Anyway, it's preferable to wait for a stable Prisma 2 and then think about it. In the mean time, we can leave this issue open..

**Comment by Errorname** [comment]
Hey @fabiofdsantos, good news! 🥳

Prisma now has a [Middleware](https://github.com/prisma/prisma-client-js/issues/770) feature, that would allow for the `sharded multi-tenancy` model to be feasible!

However, I'll first need to implement the "one DB for all tenants" model (see [issue](https://github.com/Errorname/prisma-multi-tenant/issues/40)). Then, I'll be able to make a mix of those two and make the Sharded model 🙂

**Comment by fabiofdsantos** [comment]
Great news! I no longer use Prisma but I expect to come back in a few months

**Comment by xpluscal** [comment]
Hey @Errorname is there any progress on this model?
